diff --git a/configure.in b/configure.in
index 1cfc2fc..eec4f8f 100644
--- a/configure.in
+++ b/configure.in
@@ -4484,6 +4484,7 @@ plugin_ipvs="no"
 plugin_irq="no"
 plugin_libvirt="no"
 plugin_load="no"
+plugin_nl80211="no"
 plugin_memory="no"
 plugin_multimeter="no"
 plugin_nfs="no"
@@ -4828,6 +4829,7 @@ AC_PLUGIN([mysql],       [$with_libmysql],     [MySQL statistics])
 AC_PLUGIN([netapp],      [$with_libnetapp],    [NetApp plugin])
 AC_PLUGIN([netlink],     [$with_libnetlink],   [Enhanced Linux network statistics])
 AC_PLUGIN([network],     [yes],                [Network communication plugin])
+AC_PLUGIN([nl80211],     [yes],                [Linux wireless statistics])
 AC_PLUGIN([nfs],         [$plugin_nfs],        [NFS statistics])
 AC_PLUGIN([nginx],       [$with_libcurl],      [nginx statistics])
 AC_PLUGIN([notify_desktop], [$with_libnotify], [Desktop notifications])
@@ -5162,6 +5164,7 @@ Configuration:
     network . . . . . . . $enable_network
     nfs . . . . . . . . . $enable_nfs
     nginx . . . . . . . . $enable_nginx
+    nl80211 . . . . . . . $enable_nl80211
     notify_desktop  . . . $enable_notify_desktop
     notify_email  . . . . $enable_notify_email
     ntpd  . . . . . . . . $enable_ntpd
diff --git a/src/Makefile.am b/src/Makefile.am
index f106fa1..aabbf4b 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -700,6 +700,16 @@ collectd_LDADD += "-dlopen" netlink.la
 collectd_DEPENDENCIES += netlink.la
 endif
 
+if BUILD_PLUGIN_NL80211
+pkglib_LTLIBRARIES += nl80211.la
+nl80211_la_SOURCES = collectd-nl80211.c
+nl80211_la_LDFLAGS = -module -avoid-version
+nl80211_la_CFLAGS = $(AM_CFLAGS) -I $(STAGING_DIR)/usr/include/libnl-tiny/
+nl80211_la_LIBADD = -lnl-tiny
+collectd_LDADD += "-dlopen" nl80211.la
+collectd_DEPENDENCIES += nl80211.la
+endif
+
 if BUILD_PLUGIN_NETWORK
 pkglib_LTLIBRARIES += network.la
 network_la_SOURCES = network.c network.h \
diff --git a/src/types.db b/src/types.db
index ed27b78..79044d2 100644
--- a/src/types.db
+++ b/src/types.db
@@ -172,3 +172,5 @@
 vs_processes		value:GAUGE:0:65535
 vs_threads		value:GAUGE:0:65535
 stations		value:GAUGE:0:256
+nl_station     connection_time:GAUGE:0:4294967295 inactive_time:GAUGE:0:4294967295 rx_bytes:GAUGE:0:4294967295 tx_bytes:GAUGE:0:4294967295 rx_packages:GAUGE:0:4294967295 tx_packages:GAUGE:0:4294967295 tx_retried:GAUGE:0:4294967295 tx_failed:GAUGE:0:4294967295 signal:GAUGE:-255:255 nl80211_mask:GAUGE:0:4294967295 nl80211_beacon_loss:GAUGE:0:4294967295 signal_avg:GAUGE:-255:255
+nl_survey      noise:GAUGE:-255:255 active:GAUGE:0:18446744073709551615 busy:GAUGE:0:18446744073709551615 extbusy:GAUGE:0:18446744073709551615  transmit:GAUGE:0:18446744073709551615 receive:GAUGE:0:18446744073709551615 inuse:GAUGE:0:1
