Index: utils/collectd/Makefile
===================================================================
--- utils/collectd/Makefile	(revision 32574)
+++ utils/collectd/Makefile	(working copy)
@@ -89,6 +89,7 @@
 	mysql \
 	netlink \
 	network \
+	nl80211 \
 	nginx \
 	ntpd \
 	nut \
@@ -282,6 +290,7 @@
 $(eval $(call BuildPlugin,netlink,netlink input,netlink,+PACKAGE_collectd-mod-netlink:ip))
 $(eval $(call BuildPlugin,network,network input/output,network))
 $(eval $(call BuildPlugin,nginx,nginx status input,nginx,+PACKAGE_collectd-mod-nginx:libcurl))
+$(eval $(call BuildPlugin,nl80211,nl80211 wifi statistics,nl80211,+PACKAGE_collectd-mod-nl80211:libnl-tiny))
 $(eval $(call BuildPlugin,ntpd,NTP daemon status input,ntpd,))
 $(eval $(call BuildPlugin,nut,UPS monitoring input,nut,@BROKEN))
 $(eval $(call BuildPlugin,olsrd,OLSRd status input,olsrd,))
